resources/waveshare_epd/epd2in7.py

```diff
@@ -0,0 +1,527 @@
+        for count in range(0, 44):        #vcom
+            self.send_data(self.gray_lut_vcom[count])
+            
+        self.send_command(0x21)							#red not use
+        for count in range(0, 42): 
+            self.send_data(self.gray_lut_ww[count])
+
+        self.send_command(0x22)							#bw r
+        for count in range(0, 42): 
+            self.send_data(self.gray_lut_bw[count])
+
+        self.send_command(0x23)							#wb w
+        for count in range(0, 42): 
+            self.send_data(self.gray_lut_wb[count])
+
+        self.send_command(0x24)							#bb b
+        for count in range(0, 42): 
+            self.send_data(self.gray_lut_bb[count])
+
+        self.send_command(0x25)							#vcom
+        for count in range(0, 42): 
+            self.send_data(self.gray_lut_ww[count])
+    
+    def init(self):
+        if (epdconfig.module_init() != 0):
+            return -1
+            
+        # EPD hardware init start
+        self.reset()
+        
+        self.send_command(0x01) # POWER_SETTING
+        self.send_data(0x03) # VDS_EN, VDG_EN
+        self.send_data(0x00) # VCOM_HV, VGHL_LV[1], VGHL_LV[0]
+        self.send_data(0x2b) # VDH
+        self.send_data(0x2b) # VDL
+        self.send_data(0x09) # VDHR
+        
+        self.send_command(0x06) # BOOSTER_SOFT_START
+        self.send_data(0x07)
+        self.send_data(0x07)
+        self.send_data(0x17)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0x60)
+        self.send_data(0xA5)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0x89)
+        self.send_data(0xA5)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0x90)
+        self.send_data(0x00)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0x93)
+        self.send_data(0x2A)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0xA0)
+        self.send_data(0xA5)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0xA1)
+        self.send_data(0x00)
+        
+        # Power optimization
+        self.send_command(0xF8)
+        self.send_data(0x73)
+        self.send_data(0x41)
+        
+        self.send_command(0x16) # PARTIAL_DISPLAY_REFRESH
+        self.send_data(0x00)
+        self.send_command(0x04) # POWER_ON
+        self.ReadBusy()
+
+        self.send_command(0x00) # PANEL_SETTING
+        self.send_data(0xAF) # KW-BF   KWR-AF    BWROTP 0f
+        
+        self.send_command(0x30) # PLL_CONTROL
+        self.send_data(0x3A) # 3A 100HZ   29 150Hz 39 200HZ    31 171HZ
+    
+        self.send_command(0X50)			#VCOM AND DATA INTERVAL SETTING			
+        self.send_data(0x57)
+        
+        self.send_command(0x82) # VCM_DC_SETTING_REGISTER
+        self.send_data(0x12)
+        self.set_lut()
```

Comment:
Might be better to replace magic numbers with defined constants throughout the codes:
```python3
PANEL_SETTING = 0xAF
PLL_CONTROL = 0x3A
VCOM_DATA_INTERVAL_SETTING = 0x57
```